lib/init.php

```diff
@@ -57,3 +57,58 @@ function gutenberg_menu() {
 	);
 }
 add_action( 'admin_menu', 'gutenberg_menu', 9 );
+
+function disable_scripts() {
+	// disable loading and enqueuing block editor scripts and styles
+	// can't be `__return_false` because this function is already registered for this filter
+	return false;
+}
+
+add_filter( 'should_load_block_editor_scripts_and_styles', 'disable_scripts' );
+
+// Add `editorScript` to Gutenberg block.json data if it's missing
+function add_core_editor_script( $metadata ) {
+	if ( ! empty( $metadata['name'] ) && str_starts_with( $metadata['name'], 'core/' ) ) {
+		$block_name = str_replace( 'core/', '', $metadata['name'] );
+		if ( ! isset( $metadata['editorScript'] ) ) {
+			if ( isset( $metadata['file'] ) && strstr( $metadata['file'], 'gutenberg') ) {
+				$metadata['editorScript'] = "file:./editor.min.js";
+			}
+		}
+	}
+
+	return $metadata;
+}
+
+add_filter( 'block_type_metadata', 'add_core_editor_script' );
+
+function get_block_importmap() {
+	$block_registry = WP_Block_Type_Registry::get_instance();
+	$scripts = wp_scripts();
+	$blocks         = array();
+
+	foreach ( $block_registry->get_all_registered() as $block_name => $block_type ) {
+		if ( ! isset( $block_type->editor_script_handles ) ) {
+			continue;
+		}
+		$imports = array();
+		foreach ( $block_type->editor_script_handles as $handle ) {
+			$spec = $scripts->registered[ $handle ];
+			$imports[] = array(
+				'handle' => $spec->handle,
+				'src' => $spec->src,
+				'ver' => $spec->ver
+			);
+		}
+		$blocks[ $block_name ] = $imports;
+	}
+	return $blocks;
+}
+
+function emit_importmap() {
+	wp_register_script( 'wp-importmap', '');
```

Comment:
Since we're calling `wp_enqueue_script()` down there, we don't need to register it at all. This line can be removed. We might only need to call the `wp_add_inline_script()` or `wp_localize_script()` after the `wp_enqueue_script`.